newsroom/agenda/service.py

```diff
@@ -0,0 +1,118 @@
+
+from superdesk.flask import abort
+from superdesk.core.resources import AsyncResourceService
+from superdesk.utc import local_to_utc
+from superdesk.utils import ListCursor
+
+
+class FeaturedService(AsyncResourceService[FeaturedResourceModel]):
+    resource_name = "agenda_featured"
+
+    async def on_create(self, docs):
+        """
+        Add UTC from/to datetimes on save.
+        Problem is 31.8. in Sydney is from 30.8. 14:00 UTC to 31.8. 13:59 UTC.
+        And because we query later using UTC, we store those UTC datetimes as
+        display_from and display_to.
+        """
+        for item in docs:
+            date = datetime.strptime(item._id, "%Y%m%d")
+            item.display_from = local_to_utc(item.tz, date.replace(hour=0, minute=0, second=0))
+            item.display_to = local_to_utc(item.tz, date.replace(hour=23, minute=59, second=59))
+        await super().on_create(docs)
+
+    async def find_one_for_date(self, for_date: datetime) -> FeaturedResourceModel | None:
+        return await self.find_one(req=None, display_from={"$lte": for_date}, display_to={"$gte": for_date})
+
+    async def get_featured_stories(self, req, lookup):
+        for_date = datetime.strptime(req.args.get("date_from"), "%d/%m/%Y %H:%M")
+        offset = int(req.args.get("timezone_offset", "0"))
+        local_date = get_local_date(
+            for_date.strftime("%Y-%m-%d"),
+            datetime.strftime(for_date, "%H:%M:%S"),
+            offset,
+        )
+        featured_doc = await self.find_one_for_date(local_date)
+        return await self.featured(req, lookup, featured_doc)
+
+    async def featured(self, req, lookup, featured):
+        """Return featured items.
+        :param ParsedRequest req: The parsed in request instance from the endpoint
+        :param dict lookup: The parsed in lookup dictionary from the endpoint
+        :param dict featured: list featured items
+        """
+        from newsroom.section_filters.service import SectionFiltersService
+
+        user = get_user_from_request(None)
+        company = get_company_from_request(None)
+        if is_events_only_access(user.to_dict(), company.to_dict()):
+            abort(403)
+
+        if not featured or not featured.get("items"):
+            return ListCursor([])
+
+        query = build_agenda_query()
+        await SectionFiltersService().apply_section_filter(query, self.section)
+        planning_items_query = nested_query(
+            "planning_items",
+            {"bool": {"filter": [{"terms": {"planning_items.guid": featured["items"]}}]}},
+            name="featured",
+        )
+        if req.args.get("q"):
+            query["bool"]["filter"].append(self.query_string(req.args["q"]))
+            planning_items_query["nested"]["query"]["bool"]["filter"].append(planning_items_query_string(req.args["q"]))
+
+        query["bool"]["filter"].append(planning_items_query)
+
+        source = {"query": query}
+        self.set_post_filter(source, req)
```

Comment:
Similar issue here, req is expected to be a ParsedRequest instance, and we're calling a function that doesn't exist on this class. Again, might need to wait for the Agenda resource to be upgraded to async search